disnake/client.py

```diff
@@ -741,22 +742,38 @@ async def login(self, token: str) -> None:
         data = await self.http.static_login(token.strip())
         self._connection.user = ClientUser(state=self._connection, data=data)
 
-    async def connect(self, *, reconnect: bool = True) -> None:
+    async def connect(
+        self, *, reconnect: bool = True, ignore_session_start_limit: bool = False
+    ) -> None:
         """|coro|
 
         Creates a websocket connection and lets the websocket listen
         to messages from Discord. This is a loop that runs the entire
         event system and miscellaneous aspects of the library. Control
         is not resumed until the WebSocket connection is terminated.
 
+        If the client has no more connects remaining for the day and ``ignore_session_start_limit`` is ``False``,
+        this will raise a :class:`` exception rather than connecting and resetting the
+        token. However, if ``ignore_session_start_limit`` is ``True``, this will connect regardless.
+
+        .. versionchanged:: 2.6
+            Added usage of :class:`SessionStartLimit` when connecting to the api.
+
+        .. versionchanged:: 2.6
+            Added the ``ignore_session_start_limit`` parameter.
```

Comment:
Should be moved to parameter

Reply:
https://github.com/DisnakeDev/disnake/pull/537/commits/9ca83e5f7b7cb47eb580d0834a3aea355eb6df44